### PR TITLE
feat(auth): implement password reset tokens

### PR DESCRIPTION
**Pull request by seanb4t**
## Summary

- Implements Epic 5 Task 5.5.1: Password Reset Token
- PasswordReset struct (ID, PlayerID, TokenHash, ExpiresAt, CreatedAt)
- GenerateResetToken() - 32 random bytes, hex-encoded, SHA256 hash for storage
- VerifyResetToken() - constant-time comparison to prevent timing attacks
- IsExpired() method for token validation
- PasswordResetRepository interface for persistence

## Security

- Uses crypto/rand for cryptographically secure random generation
- Stores only SHA256 hashes (never plaintext tokens)
- Uses crypto/subtle.ConstantTimeCompare to prevent timing attacks
- 1-hour expiration constant

## Test Plan

- [x] Token generation tests verify uniqueness and format
- [x] Verification tests cover correct/incorrect/empty tokens
- [x] Expiry tests cover edge cases
- [x] Linting passes
- [x] 92.3% test coverage
- [x] Code review completed, timing attack vulnerability fixed

Closes beads-dwk.21

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by seanb4t** [comment]
## Code Review Summary

Reviewed using specialized agents for code quality, error handling, type design, tests, and comments.

### Critical Issue

**`VerifyResetToken` should return `(bool, error)` to match project patterns**

The function currently returns only `bool`, making it impossible to distinguish between:
- Token verification failed (wrong token)
- Empty token provided (invalid input)
- Empty hash provided (database/state corruption)

This breaks the pattern established by `Argon2idHasher.Verify` in the same package and prevents security monitoring from distinguishing attack patterns.

**Suggested fix:**
```go
func VerifyResetToken(token, hash string) (bool, error) {
    if token == "" {
        return false, oops.Code("RESET_TOKEN_EMPTY").Errorf("reset token cannot be empty")
    }
    if hash == "" {
        return false, oops.Code("RESET_HASH_EMPTY").Errorf("stored hash cannot be empty")
    }
    computed := hashResetToken(token)
    return subtle.ConstantTimeCompare([]byte(computed), []byte(hash)) == 1, nil
}
```

### Important Issues

1. **No construction validation** - `PasswordReset` struct can be created with invalid state (empty `TokenHash`, zero `PlayerID`, expired `ExpiresAt`). Consider adding a `NewPasswordReset()` constructor.

2. **`GetByPlayer` comment ambiguity** - Says "latest" but doesn't clarify how "latest" is determined or if multiple resets per player are allowed.

### Minor Suggestions

- Add `IsExpiredAt(t time.Time)` variant for better testability
- Document `DeleteExpired` return value (count of deleted records)
- Test name "expired when ExpiresAt is exactly now" is misleading - actually tests `now.Add(-time.Nanosecond)`

### Strengths ✅

- Correct use of `crypto/rand` for secure random generation
- SHA256 hashing with constant-time comparison
- Good test coverage (92.3%) with edge cases
- Follows project error handling patterns in `GenerateResetToken`

---
🤖 Review by Claude Code

File: internal/auth/reset.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		return "", "", oops.Code("RESET_TOKEN_GENERATE_FAILED").Wrap(err)
+	}
+
+	token = hex.EncodeToString(tokenBytes)
+	hash = hashResetToken(token)
+
+	return token, hash, nil
+}
+
+// VerifyResetToken checks if the plaintext token matches the stored hash.
```
Comment (seanb4t) [inline internal/auth/reset.go:86]: **Critical:** This should return `(bool, error)` to match the `Argon2idHasher.Verify` pattern in this package. Returning only `bool` hides important failure modes (empty token vs wrong token vs corrupted hash). See summary comment for suggested implementation.

File: internal/auth/reset.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/samber/oops"
+)
+
+// Reset token configuration.
+const (
+	ResetTokenBytes  = 32        // 32 bytes = 64 hex chars
+	ResetTokenExpiry = time.Hour // 1 hour expiry
+)
+
+// PasswordReset represents a password reset request.
```
Comment (seanb4t) [inline internal/auth/reset.go:25]: Consider adding a constructor `NewPasswordReset()` that validates `playerID` is not zero, `tokenHash` is not empty, and `ttl` is positive. This prevents invalid instances from reaching the database.

File: internal/auth/reset.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+// hashResetToken computes the SHA256 hash of a token.
+func hashResetToken(token string) string {
+	h := sha256.Sum256([]byte(token))
+	return hex.EncodeToString(h[:])
+}
+
+// PasswordResetRepository manages password reset persistence.
+type PasswordResetRepository interface {
```
Comment (seanb4t) [inline internal/auth/reset.go:108]: Minor: Consider clarifying what "latest" means - ordered by `CreatedAt`? Or is only one reset per player allowed?

**Review by seanb4t — COMMENTED** [review]
## PR Review Summary

### Important Issue (1 found)

**Missing error context in `GenerateResetToken`** (`reset.go:68-74`)

The `rand.Read` error is wrapped with a code but lacks `.With()` context for debugging. Per project conventions:

```go
// Current
return "", "", oops.Code("RESET_TOKEN_GENERATE_FAILED").Wrap(err)

// Recommended  
return "", "", oops.Code("RESET_TOKEN_GENERATE_FAILED").
    With("operation", "crypto/rand.Read").
    With("requested_bytes", ResetTokenBytes).
    Wrap(err)
```

This helps operators diagnose entropy/system issues in production.

### Suggestions (optional)

1. **Add test verifying token is valid hex** (5/10 criticality) - Would catch encoding regressions
2. **Add test verifying hash determinism** (4/10 criticality) - Documents the SHA256 invariant

### Strengths ✓

- Security-conscious: `crypto/rand`, SHA256, `subtle.ConstantTimeCompare`
- Proper error codes with distinct `oops.Code` values
- 92.3% test coverage with good edge cases
- Clean `(bool, error)` API for `VerifyResetToken`
- Testable design with `IsExpiredAt()`
- Follows existing `hasher.go` patterns

---
🤖 Generated with Claude Code
